p2p/host/hostv2/hostv2.go

```diff
@@ -114,7 +114,6 @@ func (host *HostV2) SendMessage(p p2p.Peer, message []byte) error {
 		return err
 	}
 
-	defer s.Close()
```

Comment:
can you give some context of this in p2p? 

Maybe the whole logic of this function is wrong: every time we send a message we open a new stream
s, err := host.h.NewStream(context.Background(), p.PeerID, ProtocolID)

without closing this - we'll have many streams open by sending Messages.